sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridClient.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Messaging.EventGrid.Models;
+
+namespace Azure.Messaging.EventGrid.Customization
+{
+    public class EventGridClient
+    {
+        private readonly ServiceRestClient _serviceRestClient;
+        private readonly ClientDiagnostics _clientDiagnostics;
+        private readonly string _hostName;
+        private const string SasKeyName = "aeg-sas-key";
+        //private const string SasTokenName = "aeg-sas-token";
+
+        /// <summary>Initalizes an instance of EventGridClient</summary>
+        protected EventGridClient()
+        {
+        }
+
+        /// <summary>Initalizes an instance of EventGridClient</summary>
+        /// <param name="endpoint">topic endpoint</param>
+        /// <param name="credential">used to connect to Azure</param>
+        public EventGridClient(Uri endpoint, AzureKeyCredential credential)
+            : this(endpoint, credential, new EventGridClientOptions())
+        {
+        }
+
+        /// <summary>Initalizes an instance of EventGridClient</summary>
+        /// <param name="endpoint">topic endpoint</param>
+        /// <param name="credential">used to connect to Azure</param>
+        /// <param name="options">configuring options</param>
+        public EventGridClient(Uri endpoint, AzureKeyCredential credential, EventGridClientOptions options)
+        {
+            Argument.AssertNotNull(credential, nameof(credential));
```

Comment:
Nit - a lot of times we'll do `credential ??= new EventGridClientOptions()` in case people pass null explicitly becuase they don't know what else to do when getting started.